packages/wdio-config/src/utils.js

```diff
@@ -47,10 +49,19 @@ export function detectBackend (options = {}) {
      * Sauce Labs
      * e.g. 50aa152c-1932-B2f0-9707-18z46q2n1mb0
      */
-    if (typeof user === 'string' && key.length === 36) {
+    const isRDC = capabilities && 'testobject_api_key' in capabilities
+    if ((typeof user === 'string' && key.length === 36) ||
+        // When SC is used a user needs to be provided and `isRDC` needs to be true
+        (typeof user === 'string' && isRDC) ||
+        // Or only RDC
+        isRDC
+    ) {
+        // For the VM cloud a prefix needs to be added, the RDC cloud doesn't have that
+        const preFix = isRDC ? '' : 'ondemand.'
```

Comment:
let's move the ondemand part to the `getSauceEndpoint` method then

Reply:
The `getSauceEndpoint` is also used in the `@wdio/sauce-service` to determine the hostname for the rest api so moving it to there will break it.
Possible solution would be:

1. keep it here
2. create a new method that would only create the rest api hostname. Question for that one is where you want to have the method. In the `@wdio/config`, or in the `@wdio/sauce-service`. My reason for putting it in the `@wdio/sauce-service` would be that it already has the endpoint there (so it could belong in that scope), but I can also imagine if you would like to have it in the `@wdio/.config` for having all the configs there. If you'd choose for the last one then I'd also like advice to remove the `getSauceRestUrl` from the `@wdio/sauce-service` to the `@wdio/config`.

Like to hear your thoughts on that

Reply:
let's keep it then